### PR TITLE
Added alt attribute to images in listing and resource file and added …

### LINKED ISSUE #307: [Bug] The thumbnail images should have proper alt-text in listing.html
## Description
In the `listing.html` file, there are two `` tags present with class `thumbnail`. The image tags do not have an `alt` attribute specified. This needs to be corrected for all the resources with suitable text in `alt` attribute.

## Reproduction

1. Go to the codebase at https://github.com/creativecommons/cc-resource-archive.
2. Open the `listing.html` file in `docs/_layouts/listing.html`.
3. Scroll until the `` tags inside `#thumbnaillist`.
4. See error.

## Expectation
The resources should have corresponding suitable `alt` attributes inside the `` tag.

## Resolution

- [x] I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by Queen-codes**
Added alt attributes to image elements and added alt text for thumbnails

## Fixes

- Fixes #307 by @Murdock9803

## Description
This pull request adds an alt attribute to the images in listing.html and resource.html. It adds descriptive alt text to the thumbnail metadata instead of placeholder text.

## Testing
I tested the code locally and inspected the image elements in the browser to ensure the alt attribute is loading correctly.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: docs/Gemfile.lock [hunk 1]
```diff
@@ -270,6 +270,7 @@ GEM
 PLATFORMS
   aarch64-linux
   arm64-darwin-23
+  x86_64-linux
```
Comment (Murdock9803) [inline docs/Gemfile.lock:273]: This change is not needed in the scope of the issue, and hence can be un-done 🙌

**Comment by Murdock9803** [comment]
Also, @Queen-codes I guess you accidentally left some checks unchecked because of formatting error in Pull Request Description. It can be seen in the `Checklist` part.

**Comment by Murdock9803** [comment]
@possumbilities do you think manually adding `image-alt` property to all the resources pages is a good idea ? It can be difficult if number of resources are way too many. But from now onwards, we can enter an `image-alt field in the `RESOURCETEMPLATE.md` so that whenever newer resources get added, the adder fills image-alt. 
In this case @Queen-codes should go on to enter `image-alt` in every resource present as of now.

File: docs/Gemfile.lock (hunk 1 above)
Comment (Queen-codes) [reply docs/Gemfile.lock]: HI @Murdock9803. Thank you for the feedback. I have removed the gemfile.lock file.

**Comment by Queen-codes** [comment]
> Also, @Queen-codes I guess you accidentally left some checks unchecked because of formatting error in Pull Request Description. It can be seen in the `Checklist` part.

corrected, thank you

**Comment by possumbilities** [comment]
Hi @Queen-codes Thanks for the contribution, and your enthusiasm! 

The `docs` directory is the output directory which is automatically generated on build and shouldn't be edited directly. The approach @Murdock9803 is talking about is the better route, which involves. adding a new field to the `.md` files themselves and then editing that file for each resource to fill out all the alt text.

Some of that work was done here: https://github.com/creativecommons/cc-resource-archive/pull/325 but was not finished.

I'm closing for now, this and would recommend going this route instead, and avoiding editing the `docs` directory directly. Feel free to open this or another PR back up when the strategy has shifted.

**Comment by Queen-codes** [comment]
> Hi @Queen-codes Thanks for the contribution, and your enthusiasm!
> 
> The `docs` directory is the output directory which is automatically generated on build and shouldn't be edited directly. The approach @Murdock9803 is talking about is the better route, which involves. adding a new field to the `.md` files themselves and then editing that file for each resource to fill out all the alt text.
> 
> Some of that work was done here: #325 but was not finished.
> 
> I'm closing for now, this and would recommend going this route instead, and avoiding editing the `docs` directory directly. Feel free to open this or another PR back up when the strategy has shifted.

hi @possumbilities thank you very much for the feedback.

**Comment by Queen-codes** [comment]
> @possumbilities do you think manually adding `image-alt` property to all the resources pages is a good idea ? It can be difficult if number of resources are way too many. But from now onwards, we can enter an `image-alt` field in the `RESOURCETEMPLATE.md` so that whenever newer resources get added, the adder fills image-alt. In this case @Queen-codes should go on to enter `image-alt` in every resource present as of now.


> Hi @Queen-codes Thanks for the contribution, and your enthusiasm!
> 
> The `docs` directory is the output directory which is automatically generated on build and shouldn't be edited directly. The approach @Murdock9803 is talking about is the better route, which involves adding a new field to the `.md` files themselves and then editing that file for each resource to fill out all the alt text.
> 
> Some of that work was done here: #325 but was not finished.
> 
> I'm closing for now, this and would recommend going this route instead, and avoiding editing the `docs` directory directly. Feel free to open this or another PR back up when the strategy has shifted.

hi @possumbilities and @Murdock9803,

in my initial PR addressing this issue, I manually added the `image.alt` field to all the `.md` files located in the `_resources` folder—there are about seven of them. The rest of the files in the folder are `html`  files.  I wanted to clarify if you meant to apply these changes to them as well, or if there are additional `.md` files I might have missed.

I understand that I need to add an `image.alt` field to the `Resourcetemplate.md` file so that the change is applied to any new resources being added. Just wanted to confirm I'm on the same page with what's expected to resolve this issue.

Thanks you for your time and support. Looking forward to hearing from you.

**Comment by possumbilities** [comment]
Hi @Queen-codes I believe you are right, the additional `.html` files are also resource files, they just happen to contain additional html in their body rather than just frontmatter which is why they are .html files. They would need their frontmatter updated as well.

@Murdock9803 I think perhaps all the `.html` files need to be `.md` files since Markdown can contain HTML, and the split file typing might be confusing? Might be worth opening and Issue about.

**Comment by Queen-codes** [comment]
thanks for the clarification and feedback @possumbilities. I'll proceed to update the `html` files as well. 

I also think opening an issue to convert the `.html` files to `.md` for consistency sounds like a good idea. That way, there is the avoidance of confusion for future contributors, since markdown can still handle all the necessary `html` elements. @Murdock9803 happy to be of any help. thank you

**Comment by Queen-codes** [comment]
hi @possumbilities . As you suggested to open up another PR when strategy shifts, I have opened another one that has the necessary changes. Looking forward to your feedback and input. Thank you

**Comment by Murdock9803** [comment]
@Queen-codes one more crucial point to remember, As I said you did an unnecessary change in `gemfile.lock` file, and the change needed to be undone. But I believe you misunderstood the statement and removed the file from the codebase. The file is important to the codebase and only the change done is needed to be undone. If there is further confusion, please feel free to ask 🙌

**Comment by Queen-codes** [comment]
thank you very much @Murdock9803 .  I have added the `gemfile.lock` to the pull request you mentioned. Do I have to add it to the other pull request I opened too? Thank you for your help

**Comment by Murdock9803** [comment]
@Queen-codes, so basically the point is, we change only the files that need to be changed. If you remove the `Gemfile.lock` file from the codebase in your PR, then after merging the PR the codebase will lose the file (`gemfile.lock`). So whichever Pull Request you want to get merged, that should have all the files unchanged, unless its needed 🙌

**Pull request closed without merging**